confidential-data-hub/storage/src/volume_type/mod.rs

```diff
@@ -61,6 +62,12 @@ impl Storage {
                     .await?;
                 Ok(self.mount_point.clone())
             }
+            Volume::Rbd => {
```

Comment:
Rbd = rados block device? I don't see anything in this PR that is specific to RBD. Maybe this should just be called `Volume::Luks` or `Volume::Encrypted` in case we support gocryptfs in the future.

It's also interesting to note that a number of storage implementations might share the same encryption scheme but have different ways to mount the volume. I wonder if at some point we should refactor the storage design to have two phases (setup device and decrypt volume) so that some of the code could be shared. In this case we aren't doing anything to setup the device so we can probably worry about it later.

Reply:
Ah, I mean RBD = raw block device.  To avoid any ambiguity, we may consider coming up with a better name.

Reply:
Maybe we should add some document for this. I think the `RBD` plugin here would mean "given a raw block device, mount it to the target path and decrypt it with given/ephemeral key".


The initial code does not care about the device setup. This also raises some open question. If CDH cares about the setup of the device, how to ensure the availability? I am not sure how kata now handles volumes if volumes are broken during runtime. Maybe we can learn something from kata.

Reply:
Maybe spell it out as `RawBlockDevice` or just call it something else to avoid confusion.

Provisioning the device can involve operations by the kata agent (i.e. mounting something from the host) and/or operations by the CDH (i.e. pulling data from cloud storage). In the first case (this PR) I think we basically have to assume that the kata agent (or more generally the caller of the CDH) has set things up correctly.

When we start implementing more sophisticated storage backends maybe it will make sense to make things a little bit more modular.